src/devices/src/virtio/net/device.rs

```diff
@@ -114,7 +127,6 @@ pub struct Net {
     rx_bytes_read: usize,
     rx_frame_buf: [u8; MAX_BUFFER_SIZE],
 
-    tx_iovec: Vec<(GuestAddress, usize)>,
     tx_frame_buf: [u8; MAX_BUFFER_SIZE],
```

Comment:
This _buffer_ was saved on `self` and reused for each packet, instead of a local stack variable, to avoid Rust slice initialization of 65k slice on each packet (degraded performance). So now this can be changed to either:

1. Repurpose it for headers only: use correct size for this repurposed buffer ( it's now _much smaller_ :+1: ). I would also rename it for clarity:
```suggestion
    tx_headers_buf: [u8; frame_hdr_len()],
```
2. You could now even drop this _cache_ member altogether since headers are small and this optimization might be overkill,
3. Since headers slice length is known at runtime, try using `std::mem::MaybeUninit`, see [its doc](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element).

Reply:
Yes, you are right. This was a left-over. However, the MaybeUninit does not solve the problem of the size right? Since we need to call `vnet_hdr_len` to figure out the actual length we are stuck with a `Vec` I think (we cannot initialize an array without a `const` length). Am I missing something?

Reply:
`sizeof(virtio_net_hdr_v1)` is a `const` since it's determined at compile time.

You just need to mark `vnet_hdr_len()` accordingly:

```diff
-pub(crate) fn vnet_hdr_len() -> usize {
+pub(crate) const fn vnet_hdr_len() -> usize {
     mem::size_of::<virtio_net_hdr_v1>()
 }
```
or replace it with a constant altogether:

```diff
-pub(crate) fn vnet_hdr_len() -> usize {
+pub(crate) const VNET_HDR_LEN: usize = mem::size_of::<virtio_net_hdr_v1>();
```